addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -307,23 +313,25 @@ export class HtmlField extends Component {
     }
     /**
      * Toggle the code view and update the UI.
-     *
-     * @param {JQuery} $codeview
      */
     toggleCodeView() {
         this.state.showCodeView = !this.state.showCodeView;
 
-        this.wysiwyg.odooEditor.observerUnactive('toggleCodeView');
-        if (this.state.showCodeView) {
-            this.wysiwyg.odooEditor.toolbarHide();
-            const value = this.wysiwyg.getValue();
-            this.props.update(value);
-        } else {
-            this.wysiwyg.odooEditor.observerActive('toggleCodeView');
+        if (this.wysiwyg) {
+            this.wysiwyg.odooEditor.observerUnactive('toggleCodeView');
+            if (this.state.showCodeView) {
+                this.wysiwyg.odooEditor.toolbarHide();
+                const value = this.wysiwyg.getValue();
+                this.props.update(value);
+            }
+            else {
+                this.wysiwyg.odooEditor.observerActive('toggleCodeView');
+            }
+        }
```

Comment:
I might be missing something but how is this:
```js
if (this.wysiwyg) {
    this.wysiwyg.odooEditor.observerUnactive('toggleCodeView');
    if (this.state.showCodeView) {
        this.wysiwyg.odooEditor.toolbarHide();
        const value = this.wysiwyg.getValue();
          this.props.update(value);
     } else {
          this.wysiwyg.odooEditor.observerActive('toggleCodeView');
     }
}
```
Different than:
```js
if (this.wysiwyg && this.state.showCodeView) {
    this.wysiwyg.odooEditor.observerUnactive('toggleCodeView');
    this.wysiwyg.odooEditor.toolbarHide();
    const value = this.wysiwyg.getValue();
    this.props.update(value);
    this.wysiwyg.odooEditor.observerActive('toggleCodeView');
}
```

Reply:
I was confused because I feared that we might never reactivate the observer. If that is not the case then this is fine.